test/e2e/e2e.go

```diff
@@ -266,10 +272,11 @@ var _ = Describe("validation", func() {
 			}, timeout, interval).Should(MatchYAML(expectedConfigMap))
 
 			By("By checking AddressPool resource and ConfigMap are deleted")
-			Eventually(func() bool {
-				err := testclient.Client.Delete(context.Background(), addresspool)
-				return errors.IsNotFound(err)
-			}, timeout, interval).Should(BeTrue(), "Failed to delete AddressPool custom resource")
+			err := testclient.Client.Delete(context.Background(), addresspool)
+			if errors.IsNotFound(err) {
```

Comment:
i don't think it's needed and we can just check that for no error as we are the one creating the resource so it should be found.